terraform/modules/cloud-front-basic-password-protection/cloudfront.function.tf

```diff
@@ -0,0 +1,57 @@
+resource "aws_cloudfront_function" "password_protection" {
+  count = var.create ? 1 : 0
+
+  name                         = var.name
+  runtime                      = "cloudfront-js-2.0"
+  publish                      = true
+  key_value_store_associations = [
+    aws_cloudfront_key_value_store.password_protection[0].arn
+  ]
+
+  code = <<-EOF
```

Comment:
Unfortunately there is no way of passing an env var into the cloudfront function code. So I had to heredoc it in straight here as I had access to the ARN of the KV store here

Reply:
Oh interesting, I was just wondering about this.